First commit of the project's files

Lexxkit opened the pull request.

Lexxkit committed:
First commit of the project's files

k0t3n commented:
Не закоммичены миграции, в README не указано что нужно создавать директории вручную.
Миграции **всегда** должны быть в репозитории если нет резкой причины этого не делать.

k0t3n commented on items/models.py:
```diff
@@ -0,0 +1,15 @@
+    description = models.TextField()
+    image = models.ImageField(upload_to=MEDIA_ITEMS_IMAGE_DIR, null=True)
+    weight = models.IntegerField()
+    price = models.DecimalField(decimal_places=2, max_digits=11)
```
Многовато, 8 было бы вполне достаточно

k0t3n commented on items/tests.py:
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
```
пустой файл

k0t3n commented on items/urls.py:
```diff
@@ -0,0 +1,7 @@
+
+urlpatterns = [
+    path('<pk>', item_detail)
+]
```
PEP8

k0t3n commented on items/management/commands/load_data.py:
```diff
@@ -0,0 +1,98 @@
+BASE_URL = 'https://raw.githubusercontent.com/stepik-a-w/drf-project-boxes/master/'
+
+
+class Command(BaseCommand):
```
load_data - не самое лучшее название для команды, так как в django уже есть стандартная loaddata

k0t3n commented on items/management/commands/load_data.py:
```diff
@@ -0,0 +1,98 @@
+            items = requests.get(BASE_URL + 'foodboxes.json')
+            reviews = requests.get(BASE_URL + 'reviews.json')
+            users = requests.get(BASE_URL + 'recipients.json')
+        except:
```
Неявно обработанные ошибки без четкой причины делает поведение кода неожидаемым. Более того, сама ошибка даже не выводится.

k0t3n commented on items/management/commands/load_data.py:
```diff
@@ -0,0 +1,98 @@
+            reviews = requests.get(BASE_URL + 'reviews.json')
+            users = requests.get(BASE_URL + 'recipients.json')
+        except:
+            raise CommandError('Не удалось получить данные.')
```
В случае ошибки с получением, например, items упадет вся команда. По заданию предполагалась обработка ожидаемых ошибок и продолжение работы в случае если они произошли.

k0t3n commented:
Команда load_data падает с ошибкой django.core.exceptions.ValidationError: ['“” value has an invalid format. It must be in YYYY-MM-DD HH:MM[:ss[.uuuuuu]][TZ] format.']

k0t3n commented on reviews/models.py:
```diff
@@ -0,0 +1,13 @@
+class Review(models.Model):
+    author = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name='reviews')
+    text = models.TextField()
+    created_at = models.DateTimeField()
```
auto_now_add

k0t3n commented on reviews/models.py:
```diff
@@ -0,0 +1,13 @@
+    status = models.CharField(max_length=16)
+
+    def __str__(self):
+        return self.author.username
```
Ты неправильно понял смысл метода - его задача дать уникальную характеристику объекту, а у пользователя может быть много отзывов. Тем более каждый раз это будет вызывать лишний запрос в БД.

k0t3n commented on reviews/tests.py:
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
```
пустой файл

k0t3n commented on reviews/views.py:
```diff
@@ -0,0 +1,3 @@
+from django.shortcuts import render
+
```
пустой файл

k0t3n commented on users/models.py:
```diff
@@ -0,0 +1,8 @@
+
+
+class User(AbstractUser):
+    middle_name = models.CharField(max_length=256, blank=True)
```
отчества может не быть, лучше null

k0t3n commented on users/models.py:
```diff
@@ -0,0 +1,8 @@
+
+class User(AbstractUser):
+    middle_name = models.CharField(max_length=256, blank=True)
+    phone_number = models.CharField(max_length=256)
```
номер телефона вряд ли бывает 256-символьным

k0t3n commented on users/tests.py:
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
```
пустой файл

k0t3n commented on users/views.py:
```diff
@@ -0,0 +1,3 @@
+from django.shortcuts import render
+
```
пустой файл

Lexxkit committed:
Improvements after reviewer's comments:
- rename DMC from `load_data` to `get_initial_data` and make over improvements (errors during execution, error with reviews population);
- changes for Item and Review models;
- PEP8 style improvements and delete empty files.
